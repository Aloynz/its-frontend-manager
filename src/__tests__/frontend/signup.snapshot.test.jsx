import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import SignUpView from '@/app/(auth)/signup/page'

it("renders signup page unchanged", async () => {
    const result = await SignUpView()
    const { container } = render(result);
    expect(container).toMatchSnapshot();
  });
