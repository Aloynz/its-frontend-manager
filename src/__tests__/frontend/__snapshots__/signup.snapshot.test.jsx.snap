// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders signup page unchanged 1`] = `
<div>
  <div
    class="flex justify-center items-center h-screen"
  >
    <div
      class="max-w-md p-8 bg-white rounded-lg shadow-lg"
    >
      <h2
        class="text-2xl font-semibold mb-7 text-center"
      >
        ITS Frontend System Sign Up
      </h2>
      <form>
        <div
          class="space-y-2"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r0:-form-item"
          >
            Your email
          </label>
          <div
            aria-describedby=":r0:-form-item-description"
            aria-invalid="false"
            id=":r0:-form-item"
            style="display: flex; align-items: center; margin-bottom: 10px; margin-top: 10px;"
          >
            <span
              aria-label="mail"
              class="anticon anticon-mail"
              role="img"
              style="margin-right: 8px;"
            >
              <svg
                aria-hidden="true"
                data-icon="mail"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                />
              </svg>
            </span>
            <input
              class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
              name="email"
              placeholder="e.g. john@doe.com"
              style="width: 500px;"
            />
          </div>
        </div>
        <div
          class="space-y-2"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r1:-form-item"
          >
            Your password
          </label>
          <div
            aria-describedby=":r1:-form-item-description"
            aria-invalid="false"
            id=":r1:-form-item"
            style="display: flex; align-items: center; margin-bottom: 10px; margin-top: 10px;"
          >
            <span
              aria-label="lock"
              class="anticon anticon-lock"
              role="img"
              style="margin-right: 8px;"
            >
              <svg
                aria-hidden="true"
                data-icon="lock"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                />
              </svg>
            </span>
            <input
              class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
              name="password"
              placeholder="e.g. iloveits123"
              type="password"
            />
          </div>
        </div>
        <div
          class="space-y-2"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r2:-form-item"
          >
            Your institution
          </label>
          <div
            aria-describedby=":r2:-form-item-description"
            aria-invalid="false"
            id=":r2:-form-item"
            style="display: flex; align-items: center; margin-bottom: 10px; margin-top: 10px;"
          >
            <span
              aria-label="bank"
              class="anticon anticon-bank"
              role="img"
              style="margin-right: 8px;"
            >
              <svg
                aria-hidden="true"
                data-icon="bank"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M894 462c30.9 0 43.8-39.7 18.7-58L530.8 126.2a31.81 31.81 0 00-37.6 0L111.3 404c-25.1 18.2-12.2 58 18.8 58H192v374h-72c-4.4 0-8 3.6-8 8v52c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-52c0-4.4-3.6-8-8-8h-72V462h62zM512 196.7l271.1 197.2H240.9L512 196.7zM264 462h117v374H264V462zm189 0h117v374H453V462zm307 374H642V462h118v374z"
                />
              </svg>
            </span>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r3:"
              aria-expanded="false"
              class="flex h-10 w-full items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1"
              data-placeholder=""
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                name="institution"
                style="pointer-events: none;"
              >
                Select Institution
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
          <select
            aria-hidden="true"
            style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
            tabindex="-1"
          />
        </div>
        <div
          style="text-align: center; padding-bottom: 15px; padding-top: 10px;"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
            style="width: 200px; border-radius: 10px;"
            type="submit"
          >
            Register
          </button>
        </div>
      </form>
      <hr
        style="border-color: gray;"
      />
      <div
        class="mt-4 flex items-center justify-center"
      >
        <p
          class="ml-2"
        >
          Have an account? 
        </p>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 px-4 py-2"
        >
          <a
            href="/signin"
          >
            Login now
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
